zerver/tests/test_auth_backends.py

```diff
@@ -2975,9 +2975,9 @@ def test_log_into_subdomain_when_token_is_malformed(self) -> None:
             'is_signup': False,
             'redirect_to': '',
         }
-        with mock.patch("logging.warning") as mock_warn:
+        with self.assertLogs(level='WARNING') as m:
             result = self.get_log_into_subdomain(data, force_token='nonsense')
-            mock_warn.assert_called_once_with("log_into_subdomain: Malformed token given: %s", "nonsense")
+        self.assertEqual(m.output, ["WARNING:root:log_into_subdomain: Malformed token given: nonsense"])
```

Comment:
I think we need to make it clear that `nonsense` is not static and can vary based on some value(`force_token` here) by using `%s` or f-strings as you did in other places.